src/ast/CMakeLists.txt

```diff
@@ -39,8 +71,27 @@ if (STATIC_LINKING)
       clangSema
       clangSerialization
       clangToolingCore)
-  llvm_map_components_to_libnames(llvm_libs bpfcodegen ipo irreader mcjit orcjit ${LLVM_TARGETS_TO_BUILD})
-  target_link_libraries(ast ${clang_libs})
+  llvm_map_components_to_libnames(llvm_libs bpfcodegen ipo irreader mcjit orcjit option ${LLVM_TARGETS_TO_BUILD})
+
+  # Workaround to ensure that the LLVM DSO is not linked
+  function(unlink_transitive_dependency targets dep_to_remove)
+    foreach(tgt ${targets})
+      get_target_property(int_link_libs ${tgt} INTERFACE_LINK_LIBRARIES)
+
+      if(NOT(int_link_libs MATCHES "-NOTFOUND$"))
+        list(REMOVE_ITEM int_link_libs "${dep_to_remove}")
+        list(JOIN int_link_libs ";" sanitized_link_libs)
+
+        set_target_properties(${tgt} PROPERTIES
+          INTERFACE_LINK_LIBRARIES "${sanitized_link_libs}")
+      endif()
+    endforeach(candidate_dep)
+  endfunction(unlink_transitive_dependency)
+
+  unlink_transitive_dependency("${all_clang_libs}" "LLVM")
+  unlink_transitive_dependency("${llvm_libs}" "LLVMTarget")
```

Comment:
This wasn't necessary on Debian, are you sure it's necessary on Alpine?

Reply:
Glad you asked, because it doesn't seem to be needed anymore. I reordered some of these statements during the cleanup stage, so maybe it was an order of operations issue with `llvm_map_components_to_libnames` or something.